exporter/splunkhecexporter/README.md

```diff
@@ -0,0 +1,53 @@
+# Splunk HTTP Event Collector (HEC) Exporter
+
+How to send metrics to a Splunk HEC endpoint. 
+
+The following configuration options are required:
+
+- `token` (no default): HEC requires a token to authenticate incoming traffic.
+- `endpoint` (no default): Splunk HEC URL.
+
+The following configuration options can also be configured:
+
+- `source` (no default): Optional Splunk source: https://docs.splunk.com/Splexicon:Source
+- `sourcetype` (no default): Optional Splunk source type: https://docs.splunk.com/Splexicon:Sourcetype
+- `index` (no default): Splunk index, optional name of the Splunk index targeted
+- `max_connections` (default: 100): Maximum HTTP connections to use simultaneously when sending data.
+- `disable_compression` (default: false): Whether to disable gzip compression over HTTP.
```

Comment:
These options currently have no effect, right? Let's remove from the README for now. We can add once the capability is implemented in the code.

Reply:
Looks like I have dead wood then. Let me clean this up, sorry.

Reply:
Most options are used when creating the http client:
```
client: &http.Client{
			Timeout: config.Timeout,
			Transport: &http.Transport{
				Proxy: http.ProxyFromEnvironment,
				DialContext: (&net.Dialer{
					Timeout:   dialerTimeout,
					KeepAlive: dialerKeepAlive,
				}).DialContext,
				MaxIdleConns:        int(config.MaxConnections),
				MaxIdleConnsPerHost: int(config.MaxConnections),
				IdleConnTimeout:     idleConnTimeout,
				TLSHandshakeTimeout: tlsHandshakeTimeout,
				TLSClientConfig: &tls.Config{
					InsecureSkipVerify: config.InsecureSkipVerify,
				},
			},
		},
```
`disable_compression` was not handled properly. I fixed it.

Reply:
I may be wrong but looking at the code `max_connections` will have no effect because we are only using one connection. I may be missing something but I do not see where do we establish multiple connections.

Reply:
I don't think you are wrong ; the client right now does not do anything to allow multiple parallel sends. I expect there was some parallel sends coming from multiple sources to the exporter.
I added a TODO to map out further work to allow more parallelism.
As it stands right now the setting is not very useful, but is in the right place for when better parallelization will be possible.